kube-client/src/client/tls.rs

```diff
@@ -55,9 +55,18 @@ pub mod rustls_tls {
         let config_builder = if let Some(certs) = root_certs {
             ClientConfig::builder().with_root_certificates(root_store(certs)?)
         } else {
-            ClientConfig::builder()
-                .with_native_roots()
-                .map_err(Error::NoValidNativeRootCA)?
+            #[cfg(feature = "webpki-roots")]
+            {
+                // Use WebPKI roots.
+                ClientConfig::builder().with_webpki_roots()
+            }
+            #[cfg(not(feature = "webpki-roots"))]
+            {
+                // Use native roots. This will panic on Android and iOS.
+                ClientConfig::builder()
+                    .with_native_roots()
```

Comment:
There are 2 other places we call `with_native_roots` in auth plugins, we probably need to do the same thing there. Possibly this feature-if-else thing can be factored out into an internal `base_client` fn.

Reply:
Trying to factor it out into a `base_client` fn, the different error mapping between the OAuth and OIDC clients made getting generic over it quite tedious at second glance and probably not worth it, unless I've missed something obvious?